frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -2448,19 +2486,49 @@ const IR::Node* TypeInference::postorder(IR::Member* expression) {
                 return expression;
             }
         }
-        if (type->is<IR::Type_Header>()) {
-            if (member == IR::Type_Header::setValid ||
-                member == IR::Type_Header::setInvalid) {
+        if (type->is<IR::Type_Header>() || type->is<IR::Type_HeaderUnion>()) {
+            // Built-in methods
+            if ((member == IR::Type_Header::setValid ||
```

Comment:
Please add an error if setValid/setInvalid is called on a header union.
```
if (type->is<IR::Type_HeaderUnion>() && (member == IR::Type_Header::setValid || member == IR::Type_Header::setInvalid)) {
    ::error(ErrorType::ERR_INVALID, "setValid and setInvalid are not defined for header unions", expression);
    return expression;
}
```

Reply:
will do.

Reply:
Code changes from code review caused this legacy test to be updated.

testdata/p4_16_errors_outputs/issue561-1.p4-stderr

Travis has passed.  Current round of changes are ready for review.